relations/controllers.py

```diff
@@ -22,4 +27,232 @@ def service_status(params: MultiDict) -> Response:
 
     Returns ``200 OK`` if the service is up and ready to handle requests.
     """
-    return {'iam': 'ok'}, HTTPStatus.OK, {}
\ No newline at end of file
+    return {'iam': 'ok'}, HTTPStatus.OK, {}
+
+
+def create_new(arxiv_id_str: str,
+               arxiv_ver: int,
+               resource_type: str,
+               resource_id: str,
+               description: str,
+               creator: str) -> Response:
+    """
+    Create a new relation for an e-print.
+
+    Parameters
+    ----------
+    arxiv_id_str: str
+        The arXiv ID of the e-print.
+
+    arxiv_ver: int
+        The version of the e-print.
+
+    resource_type: str
+        The type of the corresponding resource.
+
+    resource_id: str
+        An identifier of the resource e.g., DOI.
+
+    description: str
+        A description for the relation.
+
+    creator: str
+        Info of the user/app who requested this relation creation.
+
+    Returns
+    -------
+    Dict[str, Any]
+        The newly-created relations, whose key is an ID,
+            and whose value is the corresponding relation in JSON format.
+        Blank if it fails.
+
+    HTTPStatus
+        An HTTP status code.
+
+    Dict[str, str]
+        A blank dict.
+
+    """
+    # get arxiv ID from str
+    arxiv_id: ArXivID = resolve_arxiv_id(arxiv_id_str)
+
+    # get relation
+    try:
+        rel: Relation = create(arxiv_id,
+                               arxiv_ver,
+                               None,
+                               RelationType.ADD,
+                               resource_type,
+                               resource_id,
+                               description,
+                               creator)
+
+        # create the result value
+        result: Dict[str, Any] = \
+            {str(rel.identifier): json.dumps(rel, default=support_json_default)}
+
+        # return
+        return result, HTTPStatus.OK, {}
+
+    except StorageError:
+        return {}, \
```

Comment:
In the controllers, for status codes >= 400, you can just raise a [Werkzeug ``HTTPException``](https://werkzeug.palletsprojects.com/en/0.15.x/exceptions/). E.g.

```python
from werkzeug.exceptions import InternalServerError

...
except StorageError as e:
    raise InternalServerError('An error occurred in storage') from e    # Nice to chain exceptions
```